src/vsphere_cpi/spec/unit/cloud/vsphere/storage_picker_spec.rb

```diff
@@ -71,47 +71,141 @@ module VSphereCloud
+            context 'and sdrs enabled on some' do
+              let(:datastore_clusters) { [sdrs_enabled_datastore_cluster1, sdrs_enabled_datastore_cluster2, sdrs_disabled_datastore_cluster] }
+              context 'along with datastores' do
+                let(:datastore_names) { ['ds-1', 'ds-2'] }
+                it 'includes a pattern constructed from datastores and datastores from all sdrs enabled datastore cluster' do
+                  expect(vm_type).to receive(:storage_policy_name).once
+                  expect(vm_type).to receive(:datastore_names).once
+                  expect(vm_type).to receive(:datastore_clusters).twice
+                  expect(vm_type).to_not receive(:datacenter)
+                  expect(global_config).to_not receive(:vm_storage_policy_name)
+                  expect(vm_type).to_not receive(:storage_policy_datastores)
+                  expect(subject).to eq(['^(ds\-1|ds\-2|sp\-1\-ds\-1|sp\-2\-ds\-1)$', nil])
+                end
+              end
+
+              context 'and no datastores' do
+                let(:datastore_names) { [] }
+                it 'includes the datastores from all sdrs enabled datastore cluster' do
+                  expect(vm_type).to receive(:storage_policy_name).once
+                  expect(vm_type).to receive(:datastore_names).once
+                  expect(vm_type).to receive(:datastore_clusters).twice
+                  expect(vm_type).to_not receive(:datacenter)
+                  expect(global_config).to_not receive(:vm_storage_policy_name)
+                  expect(vm_type).to_not receive(:storage_policy_datastores)
+                  expect(subject).to eq(['^(sp\-1\-ds\-1|sp\-2\-ds\-1)$', nil])
+                end
+              end
+            end
+            context 'and sdrs disabled on all and no datastores' do
+              let(:datastore_clusters) { [sdrs_disabled_datastore_cluster] }
+              let(:datastore_names) { [] }
+              it 'should be empty' do
+                expect(vm_type).to receive(:storage_policy_name).once
+                expect(vm_type).to receive(:datastore_names).once
+                expect(vm_type).to receive(:datastore_clusters).thrice
+                expect(vm_type).to_not receive(:datacenter)
+                expect(global_config).to_not receive(:vm_storage_policy_name)
+                expect(vm_type).to_not receive(:storage_policy_datastores)
+                expect(subject).to eq(['^()$', nil])
+              end
             end
           end
         end
-
-        context 'and sdrs disabled on all and no datastores' do
-          let(:datastore_clusters) { [sdrs_disabled_datastore_cluster] }
+        context 'with NO datastore entities defined in vm_type' do
           let(:datastore_names) { [] }
-          it 'should be empty' do
-            expect(subject).to eq('^()$')
+          let(:datastore_clusters) { [] }
+          context 'with Global Storage Policy and global datastore pattern is defined' do
```

Comment:
nitpick 'when Global Storage Policy and global datastore pattern are defined'

Reply:
The whole spec file is defined with `with` rather than `when` so letting it be.